widgets/common/src/NodeFilter.tsx

```diff
@@ -120,6 +132,22 @@ export default class NodeFilter extends React.Component {
         this.handleInputChange({}, event, field);
     };
 
+    optionsReducer(result: Record<string, string>, item: Partial<ResponseItem>): Record<string, string> {
+        let allowedOptions: string[] = [];
+
+        if (this.isFilteringSetFor(optionsField)) {
+            allowedOptions = this.getAllowedOptionsFor(optionsField);
+        }
+
+        if (allowedOptions.length !== 0 && !allowedOptions.includes(item)) {
```

Comment:
Shouldn't this be

```suggestion
        if (allowedOptions.length !== 0 && !allowedOptions.includes(item.id)) {
```

since `string[]` will never contain an object (`Partial<ResponseItem>`) :thinking: In the previous logic there was an intersection of `ids` (`string[]`) with the options, so IMO we should change it :slightly_smiling_face:

Reply:
Right, fixed